executor/envd/service/process_service.py

```
@@ -184,29 +191,45 @@ async def Start(self, request: process_pb2.StartRequest):
             if request.HasField("pty"):
-                # PTY mode - use pty
-                import pty
+                # PTY mode - use platform abstraction for cross-platform support
+                pty_manager = get_pty_manager()
 
-                master, slave = pty.openpty()
+                if not pty_manager.is_available():
+                    raise ConnectError(
+                        code="unavailable",
+                        message="PTY mode is not available on this platform",
+                    )
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)